File: ConversationIDE/src/VanEngine.Core/VAN/juul_mask.py
from __future__ import annotations

from enum import Flag, auto


class JuulMask(Flag):
    NONE = 0
    SPOKE0 = auto()
    SPOKE60 = auto()
    SPOKE120 = auto()
    SPOKE180 = auto()
    SPOKE240 = auto()
    SPOKE300 = auto()
    OUTER_RIM = auto()

File: src/VanEngine.Core/VAN/JuulMask.cs
namespace VanEngine.Core.VAN;

[Flags]
public enum JuulMask : byte
{
    None     = 0b000000,
    Spoke0   = 1 << 0,
    Spoke60  = 1 << 1,
    Spoke120 = 1 << 2,
    Spoke180 = 1 << 3,
    Spoke240 = 1 << 4,
    Spoke300 = 1 << 5,
    OuterRim = 1 << 6
}
